### PR TITLE
Js for guest

### PR DESCRIPTION
**Pull request by sarinalalria**

### CODE REVIEW

**Comment by sarinalalria** [comment]
I have to delete the copied code. If I try to resolve code now, it says to merge to complete, but I need your guys' approval before I merge it. So Once, I have your guys' approval, I will fix it the merge conflict.

**Review by Sinclair198 — APPROVED** [review]
The guest page works as intented

**Review by YiShianLiu — APPROVED** [review]

**Review by nscheiber — APPROVED** [review]

**Pull request merged**
